Sources/SwiftParser/StringLiterals.swift

```diff
@@ -598,3 +598,105 @@ public extension RawTriviaPiece {
     }
   }
 }
+
+// MARK: - StringLiteralExprSyntax/contentValue
+
+extension StringLiteralExprSyntax {
+
+  /// Returns the string value of the literal as the parsed program would see
+  /// it: Multiline strings are combined into one string, escape sequences are
+  /// resolved.
+  ///
+  /// Returns nil if the literal contains interpolation segments.
+
+  public var contentValue: String? {
```

Comment:
We need to be very very careful about trafficking in `String`s in the syntax tree. The documentation should be very clear about the limitations of this API with respect to the caveats in its implementation and the [source fidelity concerns laid out here](https://github.com/apple/swift-syntax/blob/ac4f8f3ee2d3866152bcd555a62cb8317ea3cb11/Sources/SwiftParser/SwiftParser.docc/ParserDesign.md?plain=1#L56).

Reply:
I had trouble coming up with a proper name.

I considered a couple of candidates like `stringValue`, `runtimeValue`, `content`, etc. The final decision for `contentValue` came from some existing code in ConvenienceInitializers.swift which basically does the opposite: Building a `StringLiteralExprSyntax` from an unescaped `String` value. This initializer uses the label `content:` for the value. Just `content` seemed to conflicting, though.

I'm also open for better names. I have the feeling `unescapedValue` only matches partly, leaving out the concatenation part, and is difficult to find. But I'll follow your advice here.

Reply:
I think the main reason why I don’t like `contentValue` is that `content` is already a term that’s used in SwiftSyntax, e.g. `StringLiteralExprSyntax` has a `contentLength` property that’s completely unrelated to `contentValue`. And that’s pretty confusing to me.

Alternative idea: What do you think about `representedLiteralValue`?